src/tab.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
+
+use crate::webview::{prompt::PromptDialog, WebView};
+use base::id::WebViewId;
+use serde::{Deserialize, Serialize};
+use servo_url::ServoUrl;
+use webrender_api::units::DeviceIntRect;
+
+/// Tab manager to handle multiple tab in a window.
+pub struct TabManager {
+    /// Current active tab id
+    active_tab_id: Option<WebViewId>,
+    /// Tab webview id -> Tab webview
+    tab: HashMap<WebViewId, WebView>,
+    /// Tab webview id -> Tab history
+    history: HashMap<WebViewId, TabHistory>,
+    /// Tab webview id -> Prompt dialog
+    prompt: HashMap<WebViewId, PromptDialog>,
+    // Prompt webview id -> Parent tab webview id
+    prompt_tab_map: HashMap<WebViewId, WebViewId>,
```

Comment:
Is it possible to define a custom webview type like panel and context menu? I feel like there are too many hashmap here.

Reply:
Do you mean to create a new `TabState` struct and encapsulate prompt and history in the `TabState`?

Reply:
Right, And I'm thinking maybe that will just need a hashmap eventually.